tests/decompile-test/decompilerunner.ts

```diff
@@ -149,5 +157,5 @@ function getOptsAsync(dependency: string) {
         return pkg.getCompileOptionsAsync()
             .then(opts => cachedOpts = opts);
     }
-    return Promise.resolve(cachedOpts);
+    return Promise.resolve(JSON.parse(JSON.stringify(cachedOpts)));
```

Comment:
The problem is each test was modifying the cached compiled options (since we pass the object directly). Until now, all tests were modifying the same properties (`opts.fileSystem["main.ts"]`, `opts.ast`, `opts.testMode` and `opts.ignoreFileResolutionErrors`), so it never caused an issue.

With the new functions, some tests use the new functions but others don't. I could have added an "`else { opts.useNewFunctions = false; }` so the option is reset for each test, but I prefer keeping the cached options clean from any modifications. The JSON parse / stringify is the simplest way to clone an object.